Sources/SpeziOnboarding/OnboardingFlow/OnboardingStepIdentifier.swift

```diff
@@ -13,38 +13,112 @@ import SwiftUI
     ///   - view: The view used to initialize the identifier.
     ///   - custom: A flag indicating whether the step is custom.
     @MainActor
-    init<V: View>(view: V, custom: Bool = false) {
-        self.custom = custom
-        var hasher = Hasher()
-        if let identifiable = view as? any Identifiable {
+    init(element: _OnboardingFlowViewCollection.Element, isCustom: Bool = false) {
+        self.isCustom = isCustom
+        self.viewType = type(of: element.view)
+        self.flowElementSourceLocation = element.sourceLocation
+        if let identifiable = element.view as? any OnboardingIdentifiable {
             let id = identifiable.id
-            hasher.combine(id)
-        } else if let identifiable = view as? any OnboardingIdentifiable {
+            self.identifierKind = .identifiable(id)
+        } else if let identifiable = element.view as? any Identifiable {
             let id = identifiable.id
-            hasher.combine(id)
+            self.identifierKind = .identifiable(id)
         } else {
-            hasher.combine(String(describing: type(of: view as Any)))
+            self.identifierKind = .viewTypeAndSourceLoc
         }
-        self.identifierHash = hasher.finalize()
     }
 
     /// Initializes an identifier using a view type.
     /// - Parameters:
     ///   - onboardingStepType: The class of the view used to initialize the identifier.
     ///   - custom: A flag indicating whether the step is custom.
-    init(onboardingStepType: any View.Type, custom: Bool = false) {
-        self.custom = custom
-        var hasher = Hasher()
-        hasher.combine(String(describing: onboardingStepType))
-        self.identifierHash = hasher.finalize()
+    init(onboardingStepType viewType: (some View).Type, isCustom: Bool = false) {
+        self.isCustom = isCustom
+        self.viewType = viewType
+        self.flowElementSourceLocation = nil
+        self.identifierKind = .viewTypeAndSourceLoc
+    }
+}
+
+
+extension OnboardingStepIdentifier: Hashable {
+    static func == (lhs: Self, rhs: Self) -> Bool {
+        switch (lhs.identifierKind, rhs.identifierKind) {
+        case (.viewTypeAndSourceLoc, .viewTypeAndSourceLoc):
+            lhs.viewType == rhs.viewType && lhs.flowElementSourceLocation == rhs.flowElementSourceLocation
+        case let (.identifiable(lhsValue), .identifiable(rhsValue)):
+            lhsValue.isEqual(rhsValue)
+        case (.viewTypeAndSourceLoc, .identifiable), (.identifiable, .viewTypeAndSourceLoc):
+            false
+        }
+    }
+    
+    func hash(into hasher: inout Hasher) {
+        switch self.identifierKind {
+        case .viewTypeAndSourceLoc:
+            hasher.combine(ObjectIdentifier(viewType))
+            if let flowElementSourceLocation {
+                hasher.combine(flowElementSourceLocation)
+            }
+        case .identifiable(let value):
+            hasher.combine(ObjectIdentifier(type(of: value)))
+            hasher.combine(value)
```

Comment:
@philippzagar btw this is intentionally changed to, for `Identifiable`/`OnboardingIdentifiable` step types, not only hash the identifier value, but now also the type of the identifier.

the previous version was kinda broken, in that it could result in unwanted collisions (ie, false positives for duplicate step identifiers) when using identifiers of different types that resulted in the same hash value.
eg: in the TestApp, there is a `ConsentDocumentIdentifiers` with values `first` and `second`, which is in some cases passed to the `onboardingIdentifier(_:)` modifier. this worked fine, until i added `.onboardingIdentifier(1)` to some other view, which caused the OnboardingNavigationPath to contain two steps with the same hashed identifier value (1).